src/Plugin/Chorus/CMakeLists.txt

```diff
@@ -1,40 +1,21 @@
+dpf_add_plugin(ZynChorus
+    TARGETS clap lv2 vst2 vst3
+    FILES_DSP
+        Chorus.cpp
+)
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR}/DPF/distrho .)
+include_directories(.)
 
-add_library(ZynChorus_lv2 SHARED ${CMAKE_SOURCE_DIR}/DPF/distrho/DistrhoPluginMain.cpp Chorus.cpp)
-add_library(ZynChorus_vst SHARED ${CMAKE_SOURCE_DIR}/DPF/distrho/DistrhoPluginMain.cpp Chorus.cpp)
+target_link_libraries(ZynChorus PRIVATE zynaddsubfx_core)
 
-set_target_properties(ZynChorus_lv2 PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
-set_target_properties(ZynChorus_lv2 PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv2")
-set_target_properties(ZynChorus_lv2 PROPERTIES RUNTIME_OUTPUT_DIRECTORY "lv2")
-set_target_properties(ZynChorus_lv2 PROPERTIES OUTPUT_NAME "ZynChorus")
-set_target_properties(ZynChorus_lv2 PROPERTIES PREFIX "")
-
-set_target_properties(ZynChorus_vst PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_VST2")
-set_target_properties(ZynChorus_vst PROPERTIES LIBRARY_OUTPUT_DIRECTORY "vst")
-set_target_properties(ZynChorus_vst PROPERTIES RUNTIME_OUTPUT_DIRECTORY "vst")
-set_target_properties(ZynChorus_vst PROPERTIES OUTPUT_NAME "ZynChorus")
-set_target_properties(ZynChorus_vst PROPERTIES PREFIX "")
-
-if(APPLE)
-    target_link_libraries(ZynChorus_lv2 zynaddsubfx_core ${OS_LIBRARIES} "-Wl,-exported_symbol,_lv2_descriptor" "-Wl,-exported_symbol,_lv2_generate_ttl")
-    target_link_libraries(ZynChorus_vst zynaddsubfx_core ${OS_LIBRARIES} "-Wl,-exported_symbol,_VSTPluginMain")
-else()
-    target_link_libraries(ZynChorus_lv2 zynaddsubfx_core ${OS_LIBRARIES})
-    target_link_libraries(ZynChorus_vst zynaddsubfx_core ${OS_LIBRARIES})
-endif()
-
-install(TARGETS ZynChorus_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynChorus.lv2/)
-install(TARGETS ZynChorus_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
-
-add_custom_command(TARGET ZynChorus_lv2 POST_BUILD
-    COMMAND ../../lv2-ttl-generator $<TARGET_FILE:ZynChorus_lv2>
-	WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lv2)
-
-add_dependencies(ZynChorus_lv2 lv2-ttl-generator)
+dpf__determine_vst3_package_architecture(vst3_arch)
```

Comment:
why is this line needed? the vst3 architecture should already be autodetected by dpf directly.
if that fails, it should be considered a bug and fixed on dpf side.

Reply:
Because I will also cross-build for Windows on Linux.

I specify a [toolchain file](https://github.com/AnClark/zyn-fusion-build/blob/new-generation/z/windows-build.cmake) when cross-building on Linux, in [my build script](https://github.com/AnClark/zyn-fusion-build/).

However, when cross-building, DPF CMake script won't get VST3 architecture properly. So I have to manually invoke `dpf__determine_vst3_package_architecture()`.

Reply:
right, I handle this on the other PR, but gets tricky to do it automatically.
see https://github.com/DISTRHO/dpf-cmake-action/blob/main/action.yml#L438

I suggest to remove this, and let's try to fix things on dpf side directly.

Reply:
Sorry for my mistake.

`dpf__determine_vst3_package_architecture()` is used for generating the VST3 plugin install path. For example:

```cmake
install(TARGETS ZynAddSubFX-vst3 LIBRARY DESTINATION ${PluginLibDir}/vst3/ZynAddSubFX.vst3/Contents/${vst3_arch}/)
```

Reply:
> I want to have the install step be part of dpf too, so eventually this wont be needed.

Great. That should be a better solution.

Maybe you can add some methods, for example, `dpf_install_plugin()`.